### PR TITLE
add Filter (by tags) command

### PR DESCRIPTION
**Pull request by jelneo**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/PersonContainsTagPredicate.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /**
+     * Evaluates this predicate on the given person.
+     *
+     * @return {@code true} if the person matches the predicate,
+     * otherwise {@code false}
+     */
+    @Override
+    public boolean test(ReadOnlyPerson person) {
```
Comment (khooroko) [inline src/main/java/seedu/address/model/person/PersonContainsTagPredicate.java]: perhaps you should make the method name more explanatory

**Review by khooroko — APPROVED** [review]
nice!

**Review by lawwman — APPROVED** [review]
Implementation looks good! Approved

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    private final List<String> tags;
+
+    public FilterCommand(List<String> tags) {
+        this.tags = tags;
+    }
+
+    @Override
+    public CommandResult executeUndoableCommand() throws CommandException {
+        requireNonNull(model);
```
Comment (jaivigneshvenugopal) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:29]: Hmm is this needed?

File: src/main/java/seedu/address/model/person/PersonContainsTagPredicate.java (hunk 1 above)
Comment (jaivigneshvenugopal) [inline src/main/java/seedu/address/model/person/PersonContainsTagPredicate.java:28]: Are leaving the method name as test?

File: src/test/java/seedu/address/logic/parser/FilterCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                FilterCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgs_returnsFilterCommand() {
+        // no leading and trailing whitespaces
+        FilterCommand expectedFilterCommand = new FilterCommand(Arrays.asList("violent", "friendly"));
+        assertParseSuccess(parser, "violent friendly", expectedFilterCommand);
+
+        // multiple whitespaces between keywords
```
Comment (jaivigneshvenugopal) [inline src/test/java/seedu/address/logic/parser/FilterCommandParserTest.java:30]: Great idea explaining the tests!

**Review by jaivigneshvenugopal — APPROVED** [review]
A much needed command. Great job jel!

**Pull request merged**